sycl/include/CL/sycl/handler.hpp

```diff
@@ -1157,22 +1173,16 @@ class __SYCL_EXPORT handler {
     extractArgsAndReqs();
   }
 
-  /// Defines and invokes a SYCL kernel function for the specified range.
-  ///
-  /// The SYCL kernel function is defined as SYCL kernel object. The kernel
-  /// invocation method has no functors and cannot be called on host.
-  ///
-  /// \param NumWorkItems is a range defining indexing space.
-  /// \param Kenrel is a SYCL kernel function.
-  template <int Dims>
-  void parallel_for(range<Dims> NumWorkItems, kernel Kenrel) {
-    throwIfActionIsCreated();
-    verifyKernelInvoc(Kenrel);
-    MKernel = detail::getSyclObjImpl(std::move(Kenrel));
-    detail::checkValueRange<Dims>(NumWorkItems);
-    MNDRDesc.set(std::move(NumWorkItems));
-    MCGType = detail::CG::KERNEL;
-    extractArgsAndReqs();
+  void parallel_for(range<1> NumWorkItems, kernel Kernel) {
+    parallel_for_impl(NumWorkItems, Kernel);
+  }
+
+  void parallel_for(range<2> NumWorkItems, kernel Kernel) {
+    parallel_for_impl(NumWorkItems, Kernel);
+  }
+
+  void parallel_for(range<3> NumWorkItems, kernel Kernel) {
+    parallel_for_impl(NumWorkItems, Kernel);
```

Comment:
Suggest replacing with a single template to avoid code duplication.

Reply:
I cannot do that because I need support use-case where users may pass either number of braced-init-list to the `parallel_for` call. In that case `range` should be convertible from the passed argument (either number or braced-init-list). It won't be convertible if I use function template, that's why I need explicitly list all those overloads (please see `handler_one_dim_range_init_list` for the `parallel_for` call examples)